src/components/oculus-touch-controls.js

```diff
@@ -320,20 +357,128 @@ module.exports.Component = registerComponent('oculus-touch-controls', {
+    buttonRanges.grip = {
+      min: controllerObject3D.getObjectByName('xr_standard_squeeze_pressed_min'),
+      max: controllerObject3D.getObjectByName('xr_standard_squeeze_pressed_max')
+    };
+    buttonObjects.grip.minX = buttonObjects.grip.position.x;
+
+    buttonMeshes.thumbstick = controllerObject3D.getObjectByName('thumbstick');
+    buttonObjects.thumbstick = controllerObject3D.getObjectByName('xr_standard_thumbstick_pressed_value');
+    buttonRanges.thumbstick = {
+      min: controllerObject3D.getObjectByName('xr_standard_thumbstick_pressed_min'),
+      max: controllerObject3D.getObjectByName('xr_standard_thumbstick_pressed_max'),
+      originalRotation: this.buttonObjects.thumbstick.rotation.clone()
+    };
+    buttonRanges.thumbstick.pressedY = buttonObjects.thumbstick.position.y;
+    buttonRanges.thumbstick.unpressedY =
+      buttonRanges.thumbstick.pressedY + Math.abs(buttonRanges.thumbstick.max.position.y) - Math.abs(buttonRanges.thumbstick.min.position.y);
+
+    buttonMeshes.trigger = controllerObject3D.getObjectByName('trigger');
+    buttonObjects.trigger = controllerObject3D.getObjectByName('xr_standard_trigger_pressed_value');
+    buttonRanges.trigger = {
+      min: controllerObject3D.getObjectByName('xr_standard_trigger_pressed_min'),
+      max: controllerObject3D.getObjectByName('xr_standard_trigger_pressed_max')
+    };
+    buttonRanges.trigger.diff = {
+      x: Math.abs(buttonRanges.trigger.max.rotation.x) - Math.abs(buttonRanges.trigger.min.rotation.x),
+      y: Math.abs(buttonRanges.trigger.max.rotation.y) - Math.abs(buttonRanges.trigger.min.rotation.y),
+      z: Math.abs(buttonRanges.trigger.max.rotation.z) - Math.abs(buttonRanges.trigger.min.rotation.z)
+    };
+
+    let btn1 = this.data.hand === 'left' ? 'x' : 'a';
+    let btn2 = this.data.hand === 'left' ? 'y' : 'b';
+
+    buttonMeshes[btn1 + 'button'] = controllerObject3D.getObjectByName(btn1 + '_button');
+    buttonObjects[btn1 + 'button'] = controllerObject3D.getObjectByName(btn1 + '_button_pressed_value');
+    buttonRanges[btn1 + 'button'] = {
+      min: controllerObject3D.getObjectByName(btn1 + '_button_pressed_min'),
+      max: controllerObject3D.getObjectByName(btn1 + '_button_pressed_max')
+    };
+
+    buttonMeshes[btn2 + 'button'] = controllerObject3D.getObjectByName(btn2 + '_button');
+    buttonObjects[btn2 + 'button'] = controllerObject3D.getObjectByName(btn2 + '_button_pressed_value');
+    buttonRanges[btn2 + 'button'] = {
+      min: controllerObject3D.getObjectByName(btn2 + '_button_pressed_min'),
+      max: controllerObject3D.getObjectByName(btn2 + '_button_pressed_max')
+    };
+
+    buttonRanges[btn1 + 'button'].unpressedY = buttonObjects[btn1 + 'button'].position.y;
+    buttonRanges[btn1 + 'button'].pressedY =
+      buttonRanges[btn1 + 'button'].unpressedY + Math.abs(buttonRanges[btn1 + 'button'].max.position.y) - Math.abs(buttonRanges[btn1 + 'button'].min.position.y);
```

Comment:
can consolidate this in a variable:

```
 var idButton1 = button1 + 'button';
```
Also for consistency use complete words. btn1 -> button1